### PR TITLE
CAM fixes for PIO2

### PR DESCRIPTION
**Pull request by jayeshkrishna**
This PR includes some fixes in CAM required for PIO2 and
adds the build script required to build HOMME on anlworkstation.

Although these fixes are not specific to PIO2, they showed up
during testing with PIO2.

[BFB]

### CODE REVIEW

**Comment by jayeshkrishna** [comment]
I tested this branch successfully on Cori with an F case (FC5AV1C-L + ne4_ne4) running on 4 nodes.

These changes have been more extensively tested in the dqwu/test_pio2_03 branch that has been used for several E3SM tests (developer test suite, integration test suite, high res runs etc)

File: components/homme/cmake/machineFiles/anlworkstation.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# anlworkstation
```
Comment (rljacob) [inline components/homme/cmake/machineFiles/anlworkstation.cmake]: Should this be part of this PR?

Comment (jayeshkrishna) [reply components/homme/cmake/machineFiles/anlworkstation.cmake]: This change was required to get HOMME tests building/running on anlworkstation. I can move it to a different PR if you want.

Comment (rljacob) [reply components/homme/cmake/machineFiles/anlworkstation.cmake]: Its fine here.  Just add a mention to the PR description.

Comment (jayeshkrishna) [reply components/homme/cmake/machineFiles/anlworkstation.cmake]: ok, done

**Pull request merged**